pkg/provisioner/controller.go

```diff
@@ -638,3 +656,48 @@ func (c *obcController) objectBucketForClaimKey(key string) (*v1alpha1.ObjectBuc
 	}
 	return ob, nil
 }
+
+func (c *obcController) handleUpdateClaim(key string, obc *v1alpha1.ObjectBucketClaim, class *storagev1.StorageClass) error {
+	ob, err := getObjectBucketFromClaimKey(key, obc, c.libClientset, defaultRetryBaseInterval, defaultRetryTimeout)
+	if err != nil {
+		log.Error(err, "getting ob failed", "key", key)
+		return err
+	}
+	if reflect.DeepEqual(ob.Spec.Endpoint.AdditionalConfigData, obc.Spec.AdditionalConfig) {
+		return nil
+	} else {
+		logD.Info("Additional config changed, hence updating OB")
+		tmp := make(map[string]string)
+
+		copyAdditionalConfig(ob.Spec.Endpoint.AdditionalConfigData, tmp)
+		clearAdditionalConfig(ob.Spec.Endpoint.AdditionalConfigData)
+		copyAdditionalConfig(obc.Spec.AdditionalConfig, ob.Spec.Endpoint.AdditionalConfigData)
+
+		err = c.provisioner.Update(ob)
+		if err != nil {
+			// Provisioner returns error, hence reverting values
+			clearAdditionalConfig(ob.Spec.Endpoint.AdditionalConfigData)
+			copyAdditionalConfig(tmp, ob.Spec.Endpoint.AdditionalConfigData)
+			log.Error(err, "updating from Provisioner failed")
+			return err
+		}
+		ob, err = updateObjectBucket(ob, c.libClientset)
+		if err != nil {
+			log.Error(err, "updating OB failed")
+			return err
+		}
```

Comment:
Is it necessary to update the ObjectBucket if the Update fails? 

From how I read the function up until this point, If the update has failed, I think there are no changes being made to the ObjectBucket. If that is the case, we should probably just not update the bucket at all. But maybe I missed something.